games/goradius/goradius/game.go

```diff
@@ -0,0 +1,291 @@
+
+	frames   = 30         // フレーム数
+	interval = 120        // 敵の追加間隔
+	enemie   = []*enemy{} // 敵のX座標とY座標
+)
+
+func NewGame() *Game {
+	game := &Game{
+		gopher: koebiten.NewImageFromFS(fsys, "gopher.png"),
+		x:      width / 2,
+		y:      height / 2,
+		scale:  1,
+	}
+	return game
+}
+
+// Game update process
+func (g *Game) Update() error {
+	ds := float32(0.05)
+	dt := float32(0.2)
+	speed := 1
+	dx := 1 * speed
+	dy := 1 * speed
+
+	// スタート画面からゲームプレイ画面に遷移
+	if koebiten.IsKeyPressed(koebiten.Key0) {
+		g.gameState = gameStatePlaying
+	}
+
+	// rotary buttonを回すとgopherが回転する
+	// キーボードを押しながら回すと拡大縮小する
+	if koebiten.IsKeyPressed(koebiten.KeyRotaryRight) {
+		if isAnyKeyboardKeyPressed() {
+			g.scale += ds
+		} else {
+			g.theta += dt
+		}
+	}
+	if koebiten.IsKeyPressed(koebiten.KeyRotaryLeft) {
+		if isAnyKeyboardKeyPressed() {
+			g.scale -= ds
+		} else {
+			g.theta -= dt
+		}
+	}
+
+	// joystickを倒すとgopherが移動する
+	if koebiten.IsKeyPressed(koebiten.KeyArrowRight) {
+		if g.x < width {
+			g.x += dx
+		}
+	}
+	if koebiten.IsKeyPressed(koebiten.KeyArrowLeft) {
+		if g.x > -5 {
+			g.x -= dx
+		}
+	}
+	if koebiten.IsKeyPressed(koebiten.KeyArrowDown) {
+		if g.y <= height {
+			g.y += dy
+		}
+	}
+	if koebiten.IsKeyPressed(koebiten.KeyArrowUp) {
+		if g.y > -5 {
+			g.y -= dy
+		}
+	}
+
+	// ビームを発射する
+	if g.beamCooldownTimer > 0 {
+		// クールダウン中はタイマーを減らす
+		g.beamCooldownTimer--
+		g.beamActive = false
+	} else if koebiten.IsKeyPressed(koebiten.Key1) {
+		// クールダウンが終わっていて、キーが押されていればビーム発射
+		g.beamEnergy++
+		if g.beamEnergy <= beamMax {
+			g.beamActive = true
+		} else {
+			// ビームエネルギーが上限に達したらクールダウン開始
+			g.beamActive = false
+			g.beamEnergy = 0
+			g.beamCooldownTimer = beamCooldown
+		}
+	} else {
+		// Key1が押されていない場合はビームを無効化
+		g.beamActive = false
+		g.beamEnergy = 0
+	}
+
+	// 敵の移動
+	for i := 0; i < len(enemie); i++ {
+		if i < len(enemie) { // 境界チェック
+			enemie[i].enemyX -= 1 // 少しずつ左へ
+
+			// 画面外に出た敵を削除
+			if enemie[i].enemyX < -gopherWidth {
+				enemie = append(enemie[:i], enemie[i+1:]...)
+				i--
+			}
+		}
+	}
+
+	return nil
+}
+
+func (g *Game) drawEnemy(e *enemy) {
+	// 敵を描画する
+	koebiten.DrawImageFS(nil, fsys, "enemy.png", e.enemyX, e.enemyY)
+	// 自機との当たり判定
+	if hitRects(g.x, g.y, g.x+gopherWidth, g.y+gopherHeight,
+		e.enemyX, e.enemyY, e.enemyX+gopherWidth, e.enemyY+gopherHeight) {
+		// 当たった場合、ゲームオーバー
+		g.gameState = gameStateGameOver
+	}
+}
+
+func (g *Game) drawTitle() {
+	// タイトル画面を描画する
+	koebiten.Println("Goradius")
+	koebiten.Println("Press key to start")
+}
+
+func (g *Game) drawGameOver() {
+	// ゲームオーバー画面を描画する
+	koebiten.Println("Game Over")
+}
+
+func (g *Game) drawGame(screen *koebiten.Image) {
+	// 自機描画
+	op := koebiten.DrawImageOptions{}
+	op.GeoM.Translate(-float32(gopherWidth)/2, -float32(gopherHeight)/2)
+	op.GeoM.Scale(g.scale, g.scale)
+	op.GeoM.Rotate(g.theta)
+	op.GeoM.Translate(float32(g.x), float32(g.y))
+	g.gopher.DrawImage(screen, op)
+
+	// スコアとエネルギー表示する
+	koebiten.Println("beam:", g.beamEnergy)
```

Comment:
I think `beamCooldownTimer` is better.
For example, it might be good to set it so that you can fire the next beam once you've counted up to 100.